libs/ardour/rt_tasklist.cc

```diff
@@ -86,13 +85,15 @@ RTTaskList::reset_thread_list ()
 	g_atomic_int_set (&_threads_active, 1);
 	for (uint32_t i = 0; i < num_threads; ++i) {
 		pthread_t thread_id;
-		int rv = AudioEngine::instance()->is_realtime () ?
-			pbd_realtime_pthread_create (PBD_SCHED_FIFO, AudioEngine::instance()->client_real_time_priority(), PBD_RT_STACKSIZE_HELP, &thread_id, _thread_run, this)
-			:
-			pbd_pthread_create (PBD_RT_STACKSIZE_HELP, &thread_id, _thread_run, this);
-
+		int rv = 1;
+		if (AudioEngine::instance()->is_realtime ()) {
+			rv = pbd_realtime_pthread_create (PBD_SCHED_FIFO, AudioEngine::instance()->client_real_time_priority(), PBD_RT_STACKSIZE_HELP, &thread_id, _thread_run, this);
+		}
+		if (rv) {
+			rv = pbd_pthread_create (PBD_RT_STACKSIZE_HELP, &thread_id, _thread_run, this);
+		}
 		if (rv) {
-			PBD::fatal << _("Cannot create thread for TaskList! (") << strerror(errno) << ')' << endmsg;
+			PBD::fatal << _("Cannot create thread for TaskList! (") << strerror(rv) << ')' << endmsg;
```

Comment:
What is the issue with using `? :` operators?

Reply:
I don't know if this was intentional, but the `? :` changed the behavior. 6.0 would fall back to a non-realtime thread when it couldn't create a realtime one. Pseudocode:
```cpp
if (!AudioEngine::instance()->is_realtime () || pbd_realtime_pthread_create())
    pthread_create();
```
Since 09aa0a3d1ad53f53eb4a7da0cde6644205b28f4d, it's immediate `PBD::fatal()` time if realtime thread creation fails (typically due to `EPERM`). This prevents Ardour from starting into a session when the user doesn't have realtime scheduling capability.